src/figuresClasses.ts

```diff
@@ -1,19 +1,76 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' |'rectangle',
+  color: 'red' | 'green' | 'blue',
+  getArea: Function,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(public color: 'red' | 'green' | 'blue', public a: number,
+    public b: number, public c: number, public shape: 'triangle' = 'triangle') {
+    const theLongest = Math.max(a, b, c);
+    const sides = [a, b, c];
+    const shorterSides = sides.filter((x) => x !== theLongest);
 
-}
+    if (theLongest >= shorterSides[0] + shorterSides[1]) {
+      throw new Error('This is not a triangle!');
+    }
+
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0) {
+      throw new Error('This is not a triangle!');
+    }
+
+    this.color = color;
+    this.a = a;
+    this.b = b;
```

Comment:
In TypeScript, when you declare a constructor parameter with an access modifier TypeScript implicitly creates a property on the class with the same name as the parameter and assigns the parameter value to that property.
So in your code, public a: number in the constructor is equivalent to writing this.a = a. You can safely remove this.